app/scripts/services/Message.js

```diff
@@ -0,0 +1,19 @@
+(function() {
+    function Message($firebaseArray) {
+        var ref = firebase.database().ref().child('messages');
+        var messages = $firebaseArray(ref);
+        
+        
+        return {
+            getByRoomId: function (roomId) {
+                var query = ref.orderByChild('roomId').equalTo(roomId);
+                var currentList = $firebaseArray(query);
```

Comment:
@droberts84 

I was having trouble accessing the messages in Firebase. I think the instructions in the checkpoint are a little misleading. I came up with the `var query` in order to find the correct reference to the database. Personally, I think either way has benefits.